File: 10M4W/ehowlqk/boj_1806.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "boj_1806.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [n, s] = input[0].split(" ").map((e) => Number(e));
const arr = input[1].split(" ").map((e) => Number(e));

let answer = n + 1;
let sum = arr[0];
let start = 0,
  end = 0;

while (start <= end && end < n) {
  // 조건 만족하는 경우
  if (sum >= s) {
    // 만약 현재 길이가 더 짧다면 정답 갱신
    if (end - start + 1 < answer) {
      answer = end - start + 1;
    }
    sum -= arr[start];
    ++start;
  } else {
    ++end;
    sum += arr[end];
  }
}

console.log(answer <= n ? answer : 0);

File: 10M4W/kmin/boj_1806.py
import sys
input = sys.stdin.readline
N, S = map(int, input().split())
arr = list(map(int, input().split()))
total = 0   # start, end 구간의 원소 합
start = 0
end = 0 
min_len = 1e9
while True:
    if total >= S:  # 현재 구간합이 S이상이므로
        min_len = min(min_len, end - start) # 최솟값 찾기
        total -= arr[start] 
        start += 1  # start 줄이기
    
    elif total < S: # 현재 구간합이 S보다 작으므로
        if end == len(arr):
            break   # 더 이상 불가능하므로 끄읏
        total += arr[end]
        end += 1

if min_len == 1e9:  # 아예 불가능
    print(0)
else:
    print(min_len)
